scene/2d/tile_map_layer.h

```diff
@@ -60,6 +60,15 @@ class TerrainConstraint {
 		return base_cell_coords < p_other.base_cell_coords;
 	}
 
+	bool operator==(const TerrainConstraint &p_other) const {
+		return bit == p_other.bit && base_cell_coords == p_other.base_cell_coords;
+	}
+
+	_FORCE_INLINE_ static uint32_t hash(const TerrainConstraint &p_val) {
+		uint32_t h = HashMapHasherDefault::hash(p_val.base_cell_coords);
+		return hash_murmur3_one_32(h, p_val.bit);
```

Comment:
This reasonable? Wanted to use `HashMapHasherDefault::hash` for this line too but iirc it only took one arg.